RecoMET/METProducers/src/METSignificanceProducer.cc

```diff
@@ -0,0 +1,108 @@
+// -*- C++ -*-
+//
+// Package:    METProducers
+// Class:      METSignificanceProducer
+//
+//
+
+//____________________________________________________________________________||
+#include "RecoMET/METProducers/interface/METSignificanceProducer.h"
+
+
+//____________________________________________________________________________||
+namespace cms
+{
+
+//____________________________________________________________________________||
+  METSignificanceProducer::METSignificanceProducer(const edm::ParameterSet& iConfig)
+  {
+    
+    std::vector<edm::InputTag> srcLeptonsTags = iConfig.getParameter< std::vector<edm::InputTag> >("srcLeptons");
+    for(std::vector<edm::InputTag>::const_iterator it=srcLeptonsTags.begin();it!=srcLeptonsTags.end();it++) {
+      lepTokens_.push_back( consumes<edm::View<reco::Candidate> >( *it ) );
+    }
+    
+    pfjetsToken_    = consumes<edm::View<reco::Jet> >(iConfig.getParameter<edm::InputTag>("srcPfJets"));
+
+    metToken_ = consumes<edm::View<reco::MET> >(iConfig.getParameter<edm::InputTag>("srcMet"));
+    pfCandidatesToken_ = consumes<edm::View<reco::Candidate> >(iConfig.getParameter<edm::InputTag>("srcPFCandidates"));
+
+    metSigAlgo_ = new metsig::METSignificance(iConfig);
+
+   produces<double>("METSignificance");
+   produces<double>("CovarianceMatrix00");
+   produces<double>("CovarianceMatrix01");
+   produces<double>("CovarianceMatrix10");
```

Comment:
@slava77 , as math::ErrorD<2> is failing in the same way, regarless of the tentative of inclusion of the above class into the classes_def.xml file, should I try something else?